### PR TITLE
Updated Windowmaker to 0.95.9

### PR DESCRIPTION
**Pull request by dvorberg**
Updated Windowmaker to 0.95.9 and added patch to fix crippling bug (src/osdep_darwin.c)

Tested on:
macOS 10.15.7 19H2
Xcode 12.0.1 12A7300

### CODE REVIEW

File: x11/windowmaker/Portfile [hunk 1]
```diff
@@ -1,8 +1,10 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
```
Comment (ryandesign) [inline x11/windowmaker/Portfile:1]: This is our standard modeline which states that indentations occur at multiples of 4 spaces, but the portfile's existing whitespace does not conform to that. Either don't add the modeline, or make your changes in two separate commits: one that adds the modeline and adjusts the whitespace to conform, and a second that makes functional changes.

File: x11/windowmaker/Portfile [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
+
 PortSystem 1.0
 
 name		windowmaker
-version		0.95.3
-revision        2
+version		0.95.9
+revision        1
```
Comment (ryandesign) [inline x11/windowmaker/Portfile:7]: Since this software is now hosted at GitHub, it would be a good idea to use the github portgroup. Besides simplifying the port, this should also fix the livecheck.

```suggestion
PortGroup	github 1.0

github.setup	wmaker windowmaker 0.95.9 wmaker-
revision        0
```

File: x11/windowmaker/Portfile [hunk 3]
```diff
@@ -18,15 +20,23 @@ long_description \
     It is also free software, with contributions being made \
     by programmers from around the world.
 
-master_sites \
-    http://windowmaker.org/pub/source/release/
+master_sites http://windowmaker.org/pub/source/release/
```
Comment (ryandesign) [inline x11/windowmaker/Portfile:23]:
```suggestion
github.tarball_from releases
```

File: x11/windowmaker/Portfile [hunk 4]
```diff
@@ -18,15 +20,23 @@ long_description \
...
 
 distname    WindowMaker-${version}
-checksums	rmd160 320f29bb930b046ebdb796dc0d8b2dfdfa551c1b \
-            sha256 b5eb88eccbf9a196a617594717797aa38f5c38d13d2ee40e914e153d09c9975a
+distfiles ${distname}.tar.gz
+checksums	rmd160 3c78acef4e535f4b2230e0e8cf5155c54359a61e \
+    sha256 f22358ff60301670e1e2b502faad0f2da7ff8976632d538f95fe4638e9c6b714 \
+    size 3397167
```
Comment (ryandesign) [inline x11/windowmaker/Portfile:29]: Specifying `distname` is unnecessary here because the value you've set it to is its default.
```suggestion
checksums	rmd160	3c78acef4e535f4b2230e0e8cf5155c54359a61e \
		sha256	f22358ff60301670e1e2b502faad0f2da7ff8976632d538f95fe4638e9c6b714 \
		size	3397167
```

File: x11/windowmaker/Portfile [hunk 5]
```diff
@@ -18,15 +20,23 @@ long_description \
...
+
 universal_variant no
 destroot.violate_mtree yes
 
+# patchfiles patch-wrlib-png.c.diff
+# This patch is not needed anymore. The original source contains preprocessor
+# statements to the effect of the patch.
```
Comment (ryandesign) [inline x11/windowmaker/Portfile:36]: Why add these three comment lines?

File: x11/windowmaker/Portfile [hunk 6]
```diff
@@ -18,15 +20,23 @@ long_description \
...
+
+patchfiles osdep_darwin.fix.diff
```
Comment (ryandesign) [inline x11/windowmaker/Portfile:38]: Let's add a link to where this patch came from.
```suggestion
# https://groups.google.com/g/wmaker-dev/c/EAvrkMRpla0
patchfiles osdep_darwin.fix.diff
```

File: x11/windowmaker/Portfile [hunk 7]
```diff
@@ -38,26 +48,20 @@ depends_lib \
...
-	foreach i $fl {
-		reinplace "s|/usr/X11R6/share|${prefix}/share|g" $i
-		reinplace "s|/usr/share|${prefix}/share|g" $i
-		reinplace "s|/usr/local/share|${prefix}/share|g" $i
-		reinplace "s|/opt/share|${prefix}/share|g" $i
-		reinplace "s|/opt/kde2/share|${prefix}/share|g" $i
-	}
-}
+configure.env   LINGUAS="bg cs da de el es et fi fr gl hr hu it ja \
+                ko ms nl no pl pt ro ru sk sv tr zh_CN"
```
Comment (ryandesign) [inline x11/windowmaker/Portfile:52]: This `configure.env` change reverts a560834f1e38aeec9e34b708baaed09366fa68e0 / 88ea58499404818b508cb19e6064716e2d2c9317, breaking compatibility with MacPorts 2.6.0 and later. There are warnings in the compile log caused by this. This change should be undone.
```suggestion
configure.env   "LINGUAS=bg cs da de el es et fi fr gl hr hu it ja \
                ko ms nl no pl pt ro ru sk sv tr zh_CN"
```

File: x11/windowmaker/Portfile [hunk 8]
```diff
@@ -38,26 +48,20 @@ depends_lib \
...
 
-configure.env   "LINGUAS=bg cs da de el es et fi fr gl hr hu it ja \
-                ko ms nl no pl pt ro ru sk sv tr zh_CN" \
-                "LIBS=-L${prefix}/lib -lintl -lX11"
 configure.args  --mandir=${prefix}/share/man \
-                --with-nlsdir=${prefix}/share/locale \
-                --with-gnustepdir=${prefix}/GNUstep/Local
+    --with-gnustepdir=${prefix}/GNUstep/Local \
+    --disable-magick
```
Comment (ryandesign) [inline x11/windowmaker/Portfile:56]: There is trailing whitespace on this line.
```suggestion
    --disable-magick
```

File: x11/windowmaker/Portfile [hunk 9]
```diff
@@ -38,26 +48,20 @@ depends_lib \
...
-                --with-nlsdir=${prefix}/share/locale \
-                --with-gnustepdir=${prefix}/GNUstep/Local
+    --with-gnustepdir=${prefix}/GNUstep/Local \
+    --disable-magick 
 
+variant debug description "Build with debugging flags set" {
+    # To keep .c/.o/.dSYM files around, set portautoclean no in macports.conf
+    # or use port -k to “keep” files around.
+    configure.args-append --enable-debug
+    configure.env-append CFLAGS=-g2 -O0
```
Comment (ryandesign) [inline x11/windowmaker/Portfile:62]: Don't manually put `CFLAGS` into `configure.env`. MacPorts already does it for you, putting the contents of `configure.cflags` into `CFLAGS`. If you want to change what it puts there, change `configure.cflags`.

`-O0` is an optimization flag. MacPorts stores its default optimization flag in `configure.optflags`, and that becomes a part of `configure.cflags`. If you want different optimization flags, override `configure.optflags`.

```suggestion
    configure.optflags -O0
    configure.cflags-append -g2
```

File: x11/windowmaker/Portfile [hunk 10]
```diff
@@ -38,26 +48,20 @@ depends_lib \
...
-	lappend fl ${worksrcpath}/WindowMaker/Defaults/WindowMaker.in \
-	               ${worksrcpath}/WPrefs.app/Menu.c ${worksrcpath}/WPrefs.app/Paths.c
-	foreach i $fl {
-		reinplace "s|/usr/X11R6/share|${prefix}/share|g" $i
-		reinplace "s|/usr/share|${prefix}/share|g" $i
-		reinplace "s|/usr/local/share|${prefix}/share|g" $i
-		reinplace "s|/opt/share|${prefix}/share|g" $i
-		reinplace "s|/opt/kde2/share|${prefix}/share|g" $i
-	}
-}
```
Comment (ryandesign) [inline x11/windowmaker/Portfile:52]: I don't think it's correct to remove all of these reinplaces. Some of these files, such as WPrefs.app/Menu.c, still contain references to some of those paths and need to have them changed; I did not check all the files.

**Review by ryandesign — COMMENTED** [review]
Thank you for the update but there are a few problems to address.

**Comment by ryandesign** [comment]
I forgot to mention that these lines appear in the build log:
```
checking libexif/exif-data.h usability... no
checking libexif/exif-data.h presence... no
checking for libexif/exif-data.h... no
configure: WARNING: header for EXIF library not found
```
The libexif port provides that header. Maybe a `port:libexif` dependency should be added.

**Comment by reneeotten** [comment]
@dvorberg are you able to address the review comments from @ryandesign in this PR? If so please do, so that we can get this PR merged otherwise it seems a candidate for closing as it has been stagnant for a long time now.

**Comment by reneeotten** [comment]
I have made the suggested (and other required) changes and just pushed this to `master`
